### ISSUE TITLE
Compiled the loaded model, but the compiled metrics have yet to be built

### ISSUE THREAD

**Issue by tuanndhust** [theAIGuysCode/yolov4-deepsort#79]
WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
W0621 18:01:15.284377 140678384588672 saving_utils.py:319]

**Comment by tuanndhust** [comment]
help me! thanks very much

**Comment by LukeRl** [comment]
Bump, I'm also having this issue.

**Comment by vedanthpadigelwar** [comment]
i am also having the same issue and when I'm running deepsort with yolov4 it is not detecting nor tracking any objects

**Comment by seawavve** [comment]
`! pip uninstall tensorflow`
`! pip install tensorflow==2.3.0`
It can be helpful. Put this snippet right before Step 5.

**Comment by ashishbhatti** [comment]
I faced the same issue, but the fix by @seawavve solved the problem.

**Comment by ayusheesimran** [comment]
> `! pip uninstall tensorflow` `! pip install tensorflow==2.3.0` It can be helpful. Put this snippet right before Step 5.

you saved me Ma'am!! Thanks a ton

**Comment by devtraldi** [comment]
>
```
! pip uninstall tensorflow
! pip install tensorflow==2.3.0
```

Very helpful. Thank you very much!

**Comment by Harunercul** [comment]
> `! pip uninstall tensorflow` `! pip install tensorflow==2.3.0` Yardımcı olabilir. Bu parçacığı 5. Adımdan hemen önce yerleştirin
 thank you very much

**Comment by saharan-mohit** [comment]
> `! pip uninstall tensorflow` `! pip install tensorflow==2.3.0` It can be helpful. Put this snippet right before Step 5.

It shows there is no version available

**Comment by KarahanSensoy** [comment]
It shows there is no version available and ERROR: Could not find a version that satisfies the requirement tensorflow==2.3.0 (from versions: 2.8.0rc0, 2.8.0rc1, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.9.0rc0, 2.9.0rc1, 2.9.0rc2, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.10.0rc0, 2.10.0rc1, 2.10.0rc2, 2.10.0rc3, 2.10.0, 2.10.1, 2.11.0rc0, 2.11.0rc1, 2.11.0rc2, 2.11.0, 2.11.1, 2.12.0rc0, 2.12.0rc1, 2.12.0, 2.12.1, 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.14.0rc0)
ERROR: No matching distribution found for tensorflow==2.3.0 **gives this error**

And when i change from 2.3.0 to others, i have that WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. model.compile_metrics will be empty until you train or evaluate the model.

**Comment by kroitlong** [comment]
What if i don't want change my version of tensorflow? The version i'm using is tensorflow-gpu 2.10.0 and the warning also occured.

**Comment by Shivanshvohra** [comment]
import socketio
import eventlet
import numpy as np
from flask import Flask
from keras.models import load_model
from keras.losses import MeanSquaredError
from keras.layers import Input, Dense
import base64
from io import BytesIO
from PIL import Image
import cv2

# Initialize the Socket.IO server
sio = socketio.Server()

# Initialize the Flask application
app = Flask(__name__)

# Speed limit for the simulation
speed_limit = 10

# Image preprocessing function
def img_preprocess(img):
    img = img[60:135, :, :]  # Crop the image
    img = cv2.cvtColor(img, cv2.COLOR_RGB2YUV)  # Convert to YUV color space
    img = cv2.GaussianBlur(img, (3, 3), 0)  # Apply Gaussian blur
    img = cv2.resize(img, (200, 66))  # Resize to the input shape expected by the model
    img = img / 255.0  # Normalize the image
    return img

# Telemetry event handler
@sio.on('telemetry')
def telemetry(sid, data):
    # Decode and preprocess the image
    image = Image.open(BytesIO(base64.b64decode(data['image'])))
    image = np.asarray(image)
    image = img_preprocess(image)
    image = np.array([image])  # Add batch dimension

    # Predict the steering angle
    steering_angle = float(model.predict(image))  # Get prediction
    speed = float(data['speed'])  # Extract current speed
    throttle = 1.0 - speed / speed_limit  # Calculate throttle
    send_control(steering_angle, throttle)  # Send control data

# Connection handler
@sio.on('connect')
def connect(sid, environ):
    print('Connected')
    send_control(0, 0)

# Send control data
def send_control(steering_angle, throttle):
    sio.emit('steer', data={
        'steering_angle': str(steering_angle),
        'throttle': str(throttle)
    })

if __name__ == '__main__':
    # Load the model with a custom object for MeanSquaredError
    model_path = 'C:\\Users\\tilak\\OneDrive\\Desktop\\behavioralcloning\\model.h5'  # Replace with the actual path
    model = load_model(model_path, custom_objects={'mse': MeanSquaredError()})
    
    # Middleware to integrate Socket.IO with Flask
    app = socketio.Middleware(sio, app)
    
    # Run the server
    eventlet.wsgi.server(eventlet.listen(('', 4567)), app)
    
    
    
    
    this is my code and i am getting the same error how do i fix this
